### PR TITLE
Ensure the syntax is compatible with PHP < 7.

### PR DESCRIPTION
**Pull request by lucasmichot**
Some part of the code introduce PHP 7 or PHP 7.1 new syntax, which breaks the compatibility.

This PR ensures that PHP versions < 7 are supported, as stated in the composer.json files.

### CODE REVIEW

File: src/Apple/Provider.php [hunk 1]
```diff
@@ -196,15 +196,17 @@ public function user()
...
     {
+        $fullName = '';
+
         if (request()->filled('user')) {
             $userRequest = json_decode(request('user'), true);
 
             if (array_key_exists('name', $userRequest)) {
                 $user['name'] = $userRequest['name'];
                 $fullName = trim(
-                    ($user['name']['firstName'] ?? '')
```
Comment (atymic) [inline src/Apple/Provider.php:205]: null coalesce is in 7.0

File: src/Line/Provider.php [hunk 2]
```diff
@@ -87,11 +88,11 @@ protected function getUserByToken($token)
     protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
-            'id'       => $user['userId'] ?? $user['sub'] ?? null,
+            'id'       => Arr::get($user, 'userId', Arr::get($user, 'sub')),
```
Comment (atymic) [inline src/Line/Provider.php:91]: as above

File: src/Microsoft/MicrosoftAvatar.php [hunk 3]
```diff
@@ -32,7 +32,7 @@ public function setResponse(Response $response)
      *
      * @return string
      */
-    public function getContentType(): string
+    public function getContentType()
```
Comment (atymic) [inline src/Microsoft/MicrosoftAvatar.php:35]: i'm pretty sure 7.0 supports this?

Comment (lucasmichot) [reply src/Microsoft/MicrosoftAvatar.php:35]: It does not match composer dependencies constraints: https://github.com/SocialiteProviders/Providers/blob/master/src/Microsoft/composer.json#L10

This package is supposed to work with `PHP 5.6`

And it is not the case now

File: src/Line/Provider.php (hunk 2 above)
Comment (lucasmichot) [reply src/Line/Provider.php:91]: Check https://github.com/SocialiteProviders/Providers/blob/master/src/Line/composer.json#L10

File: src/Apple/Provider.php (hunk 1 above)
Comment (lucasmichot) [reply src/Apple/Provider.php:205]: Same as other comments @atymic, all those providers are supposed to work with `PHP 5.6`...

Introducing PHP 7 or 7.1 specific syntax breaks this compatibility.

Comment (atymic) [reply src/Apple/Provider.php:205]: Ok, feel free to drop < 7 then. We really need to drop < 7.2 some time soon as well.

**Comment by lucasmichot** [comment]
Hey @atymic, thanks for your feedback.
I am happy to provide a PR that drops support for PHP < 7

> We really need to drop < 7.2 some time soon as well.

Why not doing it now directly then?

Is this something part of your backlog (if so, I could not see it)
cc / @m1guelpf

**Comment by atymic** [comment]
> I am happy to provide a PR that drops support for PHP < 7

I'd just drop it for the ones using syntax that doesn't work on lower versions, then when we drop < 7.2 we can do all of them. 

>Why not doing it now directly then?
> Is this something part of your backlog (if so, I could not see it)

Feel free to create a PR.  7.0 hasn't been supported for almost a two and a half years, and 7.1 for one and a half.
https://www.php.net/supported-versions.php

**Comment by atymic** [comment]
@lucasmichot 

I've merged your other PRs, thanks for all the awesome work 😀 
How about you PR dropping support for < `7.2` and then I'll make a new major version release of all providers?

Feel free to suggest any other breaking changes while we're at it :)

**Pull request closed without merging**